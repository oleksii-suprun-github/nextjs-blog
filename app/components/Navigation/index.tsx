'use client';

import { MouseEventHandler } from 'react';
import { usePathname } from 'next/navigation';
import Link from 'next/link';
import clsx from 'clsx';

const Navigation = ({ onClick = () => {} }: { onClick?: MouseEventHandler<HTMLAnchorElement> }) => {
  const path = usePathname();
  const isActiveLink = (href: string) => path === href;

  const navigationLinks = [
    { uid: 'home', label: 'Home', href: '/' },
    { uid: 'about', label: 'About', href: '/about' },
    { uid: 'publications', label: 'Publications', href: '/publications' },
  ];

  return (
    <div className="mb-12 mt-16 flex flex-col items-center lg:mb-0 lg:mt-0 lg:flex-row lg:items-start">
      {navigationLinks.map((link) => (
        <Link
          className={clsx(
            'mb-5 text-xl lg:mb-0 lg:mr-12 lg:text-lg',
            isActiveLink(link.href) && 'font-bold',
          )}
          key={link.uid}
          href={link.href}
          onClick={onClick}
        >
          <span className="text-stone-300 hover:text-stone-100">{link.label}</span>
        </Link>
      ))}
    </div>
  );
};

export default Navigation;
