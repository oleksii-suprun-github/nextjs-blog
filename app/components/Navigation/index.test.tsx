import { render } from '@testing-library/react';
import { usePathname } from 'next/navigation';
import { vi, Mock } from 'vitest';
import Navigation from '.';

vi.mock('next/navigation', () => ({
  usePathname: vi.fn(),
}));

describe('Navigation Component', () => {
  it('renders correctly when no link is active', () => {
    (usePathname as Mock).mockReturnValue('/');

    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly when journal link is active', () => {
    (usePathname as Mock).mockReturnValue('/journal');

    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly when statistics link is active', () => {
    (usePathname as Mock).mockReturnValue('/statistics');

    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });
});
