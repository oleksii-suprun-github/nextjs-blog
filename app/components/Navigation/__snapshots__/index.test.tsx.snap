// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation Component > renders correctly when journal link is active 1`] = `
<div>
  <div
    class="mb-12 mt-16 flex flex-col items-center lg:mb-0 lg:mt-0 lg:flex-row lg:items-start"
  >
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        Home
      </span>
    </a>
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/about"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        About
      </span>
    </a>
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/publications"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        Publications
      </span>
    </a>
  </div>
</div>
`;

exports[`Navigation Component > renders correctly when no link is active 1`] = `
<div>
  <div
    class="mb-12 mt-16 flex flex-col items-center lg:mb-0 lg:mt-0 lg:flex-row lg:items-start"
  >
    <a
      class="mb-5 text-2xl font-bold lg:mb-0 lg:mr-12 lg:text-base"
      href="/"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        Home
      </span>
    </a>
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/about"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        About
      </span>
    </a>
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/publications"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        Publications
      </span>
    </a>
  </div>
</div>
`;

exports[`Navigation Component > renders correctly when statistics link is active 1`] = `
<div>
  <div
    class="mb-12 mt-16 flex flex-col items-center lg:mb-0 lg:mt-0 lg:flex-row lg:items-start"
  >
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        Home
      </span>
    </a>
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/about"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        About
      </span>
    </a>
    <a
      class="mb-5 text-lg lg:mb-0 lg:mr-12"
      href="/publications"
    >
      <span
        class="text-stone-300 hover:text-stone-100"
      >
        Publications
      </span>
    </a>
  </div>
</div>
`;
